### PR TITLE
feat: generate and publish types

### LINKED ISSUE #279: Generate and publish types
As we're pretty much typed now, I think we should generate and publish types when we publish `15.0.0`.

This would require a setup similar to: https://github.com/voxpelli/pony-cause/blob/1a93b683ef5843ca43a67d1953460863372f4c73/package.json#L18-L20

### PR DESCRIPTION
**Pull request by Divlo**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Generate and publish types.

**Which issue (if any) does this pull request address?**

Fixes #279 

**Is there anything you'd like reviewers to focus on?**

Similar setup to https://github.com/voxpelli/pony-cause

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,28 +60,29 @@
...
     "npm-run-all": "^4.1.5",
-    "standard": "*",
+    "standard": "^17.0.0-2",
     "tape": "^5.0.1",
+    "type-coverage": "^2.21.0",
     "typescript": "~4.5.2"
   },
   "engines": {
-    "node": "^12.20.0 || ^14.13.1 || >=16.0.0"
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
```
Comment (LinusU) [inline package.json:85]: Is this a breaking change? 🤔

Comment (voxpelli) [reply package.json:85]: Since we're in a `17.0.0` pre-release: No
Otherwise: Yes

File: test/api.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const { StandardEngine } = require('../')
 
 async function getStandard () {
   /** @type {string} */
-  const configFile = (await import('../tmp/standard/options.js')).default.eslintConfig.configFile
+  const configFile = (await import('../tmp/standard/options.js')).default.eslintConfig.overrideConfigFile
```
Comment (voxpelli) [inline test/api.js:9]: This is unrelated to this PR, right? It's to restore tests to work for latest `standard`, right? Ideally should be a separate PR, but lets deal with it here

File: package.json [hunk 3]
```diff
@@ -53,28 +60,29 @@
...
+    "eslint-config-standard": "^17.0.0-1",
+    "eslint-config-standard-jsx": "^11.0.0-1",
+    "eslint-plugin-import": "^2.25.4",
+    "eslint-plugin-n": "^14.0.0",
+    "eslint-plugin-promise": "^6.0.0",
     "eslint-plugin-react": "^7.28.0",
     "installed-check": "^5.0.0",
     "npm-run-all": "^4.1.5",
-    "standard": "*",
+    "standard": "^17.0.0-2",
```
Comment (voxpelli) [inline package.json:79]: This is unrelated to this PR, right? It's to restore tests to work for latest standard, right? Ideally should be a separate PR, but lets deal with it here

**Review by voxpelli — APPROVED** [review]
Looks good in general

**Review by LinusU — APPROVED** [review]

File: test/api.js (hunk 2 above)
Comment (Divlo) [reply test/api.js:9]: Yes, it is to restore tests to work for latest `standard`, tests would fail otherwise.
